mlflow/store/artifact/azure_blob_artifact_repo.py

```diff
@@ -29,7 +30,18 @@ def __init__(self, artifact_uri, client=None):
         from azure.storage.blob import BlobServiceClient
 
         (_, account, _) = AzureBlobArtifactRepository.parse_wasbs_uri(artifact_uri)
-        if "AZURE_STORAGE_CONNECTION_STRING" in os.environ:
+        if os.getenv("MLFLOW_USE_AZURE_IDENTITY", "False").lower() is "true":
+            from azure.identity import DefaultAzureCredential
```

Comment:
I have added the dependency within the if clause to avoid adding a global dependency to this class.

Reply:
Don't use `is` to compare string equality.

Reply:
Oh wow, not sure how this made it here. I guess the lint tests will have catch that anyway? I will fix it now

Reply:
@alanjcastonguay I have address your comment. Would you mind reviewing the implementation?